apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/network_constraints.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2016 Cisco Systems Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import netaddr
+import os
+
+from oslo_config import cfg
+from oslo_log import log as logging
+
+
+LOG = logging.getLogger(__name__)
+
+SCOPE_PUBLIC = 'public'
+SCOPE_PRIVATE = 'private'
+SCOPE_DENY = 'deny'
+
+
+class NetworkConstraintsSource(object):
+    def __init__(self):
+        pass
+
+    def get_subnet_constraints(self, tenant, network):
+        """Returns pair (default-subnet-scope, scope-dict-for-network) """
+        return (None, None)
+
+
+class NetworkConstraints(object):
+    def __init__(self, constraints_source):
+        self.source = constraints_source
+
+    def get_subnet_scope(self, tenant, network, cidr):
+        scope = None
+        cidr = netaddr.IPNetwork(cidr)
+        if self.source:
+            def_scope, constraints = (
+                self.source.get_subnet_constraints(tenant, network))
+            scope = def_scope or scope
+            if constraints:
+                if cidr in (constraints.get('deny') or netaddr.IPSet()):
```

Comment:
So if I'm creating a subnet with CIDR 192.168.0.0/16, and 192.168.1.0/24 is in the "deny" list for that tenant-network, would my subnet get the default scope? I would assume it gets rejected as overlapping with the latter.